tsdb/head.go

```diff
@@ -1762,7 +1762,9 @@ func newMemSeries(lset labels.Labels, id uint64, chunkRange int64) *memSeries {
 }
 
 func (s *memSeries) minTime() int64 {
-	if len(s.chunks) == 0 {
+	// TODO(beorn7): s.chunks[0] == nil has been observed in the wild but it
+	// is not clear if its occurrence is regular or a bug.
+	if len(s.chunks) == 0 || s.chunks[0] == nil {
```

Comment:
This can never be `nil`. There is only 1 place where chunks are added to the slice https://github.com/prometheus/prometheus/blob/4d6f9b75e230465443391ca76a4182b130eb99e9/tsdb/head.go#L1780-L1785

As you said, it won't be good to mask this bug. I will take a deeper look at this (and also isolation work in general) on Monday.

Reply:
We also shrink that. I am more thinking about the fact that we might need some locking about s.chunks

https://github.com/prometheus/prometheus/blob/4d6f9b75e230465443391ca76a4182b130eb99e9/tsdb/head.go#L1840-L1848

Reply:
See full issue #6977

Reply:
Yup, but shrinking also doesn't seem to introduce `nil` chunks. And about locking, we have locks around the series itself.